src/server/scripts/fsq_ingestion.ts

```diff
@@ -0,0 +1,151 @@
+import axios from "axios";
+
+import env from "@/config/env.config";
+import { InsertPlace, places } from "@/server/db/schema";
+
+import { db } from "../db";
+
+export interface Feature {
+  geometry: {
+    coordinates: [number, number];
+  };
+  properties: {
+    id: string;
+  };
+}
+
+export interface PlaceInfo {
+  fsq_id: string;
+  name: string;
+  categories: { name: string; id: number }[];
+  location: { address: string };
+  distance: string;
+  rating: string;
+  photos: {}[];
+  geocodes: { main: { latitude: number; longitude: number } };
+  features: {};
+}
+
+const FOURSQUARE_BASE_URL: string = "https://api.foursquare.com/v3/places/search";
+
+function checkCoordinatesAndCategory(item: PlaceInfo) {
+  return item.geocodes?.main?.latitude && item.geocodes?.main?.longitude && item.categories?.[0]?.name && item.fsq_id;
+}
+const categories =
+  "10001,10002,10004,10007,10013,10024,10027,10056,12009,12055,12056,12061,12060,12059,12058,12057,12075,12071,12072,12070,12098,12124,13003,13029,13032,13035,13036,13034,13040,13040,13052,13065,13064,14001,14002,14004,15000,15005,15014,16000,16001,16023,16032,16037,18000,19000,13025,11128,16005,12004,13019,18021,13027,13018,13054,12080,16019,18028,10032,17114,18025,17020,13059,13002,13046,13022";
+
+function createPlace(placeInfo: PlaceInfo, feature: Feature): InsertPlace {
+  return {
+    fsq_id: placeInfo.fsq_id,
+    property_ids: [feature.properties.id],
+    name: placeInfo.name,
+    category_name: placeInfo.categories[0].name,
+    category_id: placeInfo.categories[0].id,
+    coordinates: {
+      x: placeInfo.geocodes.main.latitude,
+      y: placeInfo.geocodes.main.longitude,
+    },
+    distance: placeInfo.distance,
+    rating: placeInfo.rating,
+    photos: placeInfo.photos,
+    address: placeInfo.location.address,
+    features: placeInfo.features,
+  };
+}
+
+async function getNearByPlaces(
+  feature: any,
+  currentPage = 1,
+  accumulatedData: InsertPlace[] = [],
+  nextUrl?: string
+): Promise<InsertPlace[]> {
+  const lat = feature.geometry.coordinates[1];
+  const lon = feature.geometry.coordinates[0];
+
+  try {
+    const response = nextUrl
+      ? await axios.get(nextUrl, {
+          headers: {
+            Authorization: env.foursquare.apiKey,
+          },
+        })
+      : await axios.get(FOURSQUARE_BASE_URL, {
+          headers: {
+            Authorization: env.foursquare.apiKey,
+          },
+          params: {
+            ll: `${lat},${lon}`,
+            radius: 3000,
```

Comment:
Will this radius give enough results?

Reply:
actually we have to change radius and ingest to get more results which is already ingested